numba/tests/test_function.py

```diff
@@ -0,0 +1,575 @@
+import types as pytypes
+from numba import njit, function, cfunc, types, int64, int32
+import ctypes
+
+from .support import TestCase
+
+
+def dump(foo):  # FOR DEBUGGING, TO BE REMOVED
+    foo_type = function.fromobject(foo)
+    foo_sig = foo_type.signature()
+    foo.compile(foo_sig)
+    print('{" LLVM IR OF "+foo.__name__+" ":*^70}')
+    print(foo.inspect_llvm(foo_sig.args))
+    print('{"":*^70}')
+
+
+# Decorators for tranforming a Python function to different kinds of functions:
+def pure_func(func, sig=int64(int64)):
+    assert isinstance(func, pytypes.FunctionType), repr(func)
```

Comment:
This code is not exercised. I would prefer not to include code that is never used.

Reply:
Removed.